src/mvt_getter.py

```diff
@@ -0,0 +1,383 @@
+# -*- coding: utf-8 -*-
+from mapcss import MapCSS
+import sys
+import math
+from optparse import OptionParser
+from mapcss import _test_feature_compatibility
+
+reload(sys)
+sys.setdefaultencoding("utf-8")  # a hack to support UTF-8
+
+mapped_cols = {}
+osm2pgsql_avail_keys = {}
+
+def escape_sql_column(name, type, asname=False):
+    if name in mapped_cols:
+        if asname:
+            return mapped_cols[name] + ' as "%s"' % name
+        else:
+            return mapped_cols[name]
+
+    name = name.strip().strip('"')
+    type = {'line': 'way', 'area': 'way'}.get(type, type)
+    if type in osm2pgsql_avail_keys.get(name, ()) or not osm2pgsql_avail_keys:
+        return '"' + name + '"'
+    elif not asname:
+        return "(tags->'" + name + "')"
+    else:
+        return "(tags->'" + name + "') as \"" + name + '"'
+
+def pixel_size_at_zoom(z, l=1):
+    """
+    Converts l pixels on tiles into length on zoom z
+    """
+    return int(math.ceil(l * 20037508.342789244 / 512 * 2 / (2 ** z)))
+
+def get_sql(condition, obj):
+    # params = [re.escape(x) for x in self.params]
+    params = condition.params
+    t = condition.type
+    if t == 'eq':   # don't compare tags against sublayers
+        if params[0][:2] == "::":
+            return ("", "")
+    try:
+        column_name = escape_sql_column(params[0], type=obj)
+
+        if t == 'eq':
+            return params[0], '%s = \'%s\'' % (column_name, params[1])
+        if t == 'ne':
+            return params[0], '(%s != \'%s\' or %s IS NULL)' % (column_name, params[1], column_name)
+        if t == 'regex':
+            return params[0], '%s ~ \'%s\'' % (column_name, params[1].replace("'", "\\'"))
+        if t == 'true':
+            return params[0], '%s = \'yes\'' % (column_name)
+        if t == 'untrue':
+            return params[0], '%s = \'no\'' % (column_name)
+        if t == 'set':
+            return params[0], '%s IS NOT NULL' % (column_name)
+        if t == 'unset':
+            return params[0], '%s IS NULL' % (column_name)
+
+        if t == '<':
+            return params[0], """(CASE WHEN %s  ~  E'^[-]?[[:digit:]]+([.][[:digit:]]+)?$' THEN CAST (%s AS FLOAT) &lt; %s ELSE false END) """ % (column_name, column_name, params[1])
+        if t == '<=':
+            return params[0], """(CASE WHEN %s  ~  E'^[-]?[[:digit:]]+([.][[:digit:]]+)?$' THEN CAST (%s AS FLOAT) &lt;= %s ELSE false END)""" % (column_name, column_name, params[1])
+        if t == '>':
+            return params[0], """(CASE WHEN %s  ~  E'^[-]?[[:digit:]]+([.][[:digit:]]+)?$' THEN CAST (%s AS FLOAT) > %s ELSE false END) """ % (column_name, column_name, params[1])
+        if t == '>=':
+            return params[0], """(CASE WHEN %s  ~  E'^[-]?[[:digit:]]+([.][[:digit:]]+)?$' THEN CAST (%s AS FLOAT) >= %s ELSE false END) """ % (column_name, column_name, params[1])
+    except KeyError:
+        pass
+
+def get_sql_hints(choosers, obj, zoom):
```

Comment:
ok, I moved `get_sql_hints` from MapCSS parser in here and I can't say I am feeling wrong about doing that. Maybe MapCSS doesn't need to know how to generate sql for my specific purpose, maybe I would want it to generate sql compatible with different OSM schema in the future, why not.